payload.go

```diff
@@ -1,29 +1,36 @@
 package exoip
 
 import (
+	"crypto/rand"
 	"encoding/hex"
 	"errors"
 	"fmt"
 	"net"
 	"strings"
 )
 
-const payloadLength = 24
+const payloadLength = 40
 
-// NewPayload builds a Payload from a raw buffer (length of 24)
+// NewPayload builds a Payload from a raw buffer (length of 40)
 //
 // The layout of the payload is as follows:
 //
-//      2bytes  2bytes  4 bytes         16 bytes
-//     +-------+-------+---------------+-------------------------------+
-//     | PROTO | PRIO  |    EIP        |   NicID (128bit UUID)         |
-//     +-------+-------+---------------+-------------------------------+
+//      2bytes  2bytes      4 bytes
+//     ┏━━━━━━━┳━━━━━━━┳━━━━━━━━━━━━━━━┓
+//     ┃ PROTO ┃ PRIO  ┃    EIP        ┃            16 bytes
+//     ┣━━━━━━━┻━━━━━━━┻━━━━━━━━━━━━━━━┻━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
+//     ┃ NicID (128bit UUID)                                          ┃
+//     ┣━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┫
+//     ┃ Message ID (128bit UUID)                                     ┃
+//     ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
 //
 func NewPayload(buf []byte) (*Payload, error) {
 	protobuf := make([]byte, 2)
 	protobuf = buf[0:2]
-	uuidbuf := make([]byte, 16)
-	uuidbuf = buf[8:24]
+	nicbuf := make([]byte, 16)
+	nicbuf = buf[8:24]
+	msgbuf := make([]byte, 16)
+	msgbuf = buf[24:40]
 
 	version := hex.EncodeToString(protobuf)
 	if ProtoVersion != version {
```

Comment:
Shouldn't it handle the case for *lower* version and avoid extracting information not present, to avoid interference while upgrading the daemon on the servers?